### PR TITLE
TechDebt Cypress v12 migration

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

* Upgraded to Cypress v12, which had breaking changes:
    * Reworked all `e2e` tests so they are all independent as per `best practices in cypress`
    * unified on `beforeEach` statements
    * Using `session` in conjunction with the login, to help performance and session management
    * Cut actual test run time down from ~55s to ~35s; Still relies on the stack to stand up though which is ~4m ☹️ 
    * executing `a11y` testing after every test now with `afterEach`
    * introduced a base `fixture` to check `state` at load; We can build from this for additional tests that rely on a specific state later on.
* Fixed a lingering `linting` issue in `auth.js` 

## Issue

* All `e2e` have been randomly failing

## How to test

`yarn test:e2e` or `yarn test:e2e:interactive`

## Links

[Migrating to Cypress 12.0+](https://docs.cypress.io/guides/references/migration-guide)

### CODE REVIEW

File: frontend/cypress/e2e/mainPage.cy.js [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+    cy.checkA11y();
 });
 
-it("has expected state on load", () => {
-    cy.visit("/");
-    cy.window()
-        .then((win) => win.store.getState().auth)
-        .should("deep.include", { isLoggedIn: false });
+it("has expected state on initial load", () => {
+    // cy.window()
```
Comment (johndeange) [inline frontend/cypress/e2e/mainPage.cy.js:11]: please remove commented code

**Review by johndeange — APPROVED** [review]
The fixture with the initial state is awesome! E2E looks a lot cleaner.

**Comment by jonnalley** [comment]
👏

File: frontend/cypress/e2e/mainPage.cy.js (hunk 1 above)
Comment (tdonaworth) [reply frontend/cypress/e2e/mainPage.cy.js]: 👍

**Pull request merged**